kcidb/db/__init__.py

```diff
@@ -134,6 +135,24 @@ def empty(self):
         assert self.is_initialized()
         self.driver.empty()
 
+    def reset(self, database=None):
+        """
+        Recreates a database driver instance on request
+        using existing database string
+        Args:
+            database:   A string specifying the database to access, formatted
+                        as "<DRIVER>:<PARAMS>" or just "<DRIVER>". Where
+                        "<DRIVER>" is the driver name, and "<PARAMS>" is the
+                        optional driver-specific database parameter string
+                        or None.
+
+        Returns:
+            A new client object
```

Comment:
No, this method should not create a new client. We can already do that ourselves. What we need to do is to keep the same `Client` object, but make it re-create the driver instance it's using (`self.driver`).

Essentially you can move the `self.driver = ...` assignment and the following assertion from `__init__()` here, and then call this method from `__init__()` instead.